packages/vm/src/context.rs

```diff
@@ -52,7 +58,28 @@ fn destroy_unmanaged_context_data<S: Storage, Q: Querier>(ptr: *mut c_void) {
 }
 
 /// Get a mutable reference to the context's data. Ownership remains in the Context.
-fn get_context_data<S: Storage, Q: Querier>(ctx: &mut Ctx) -> &mut ContextData<S, Q> {
+// NOTE: This is actually not really implemented safely at the moment. I did this as a
```

Comment:
Since iterator retains a reference to storage, it is impossible to release it into the raw pointer / unbounded lifetime of ContextData, as the compiler can never make a guarantee of when storage is released relative to iterators. Thus, we need to take a holistic view of the program and do this "unsafe" management, ensuring that we do understand the lifetime.

I took care in the `ContextData` section to ensure that all iterators are released before the storage is returned. I would be very happy for an eagle eye over all possible usages of the `ContextData` assuming it happens through a function defined in this module. 

The relative lifetimes of storage/iterators is independent of whether or not they are go pointers or local rust "MockStorage". Just want to separate out these two issues. Happy for a better lifetime solution, but I doubt there is one without unsafe, as we need information that compiler cannot prove.

Reply:
Can you please open another issue so we can track this concern, and then we'll resolve this discussion for now?

Reply:
Here we go: https://github.com/CosmWasm/cosmwasm/issues/335

Reply:
thanks!